nav2_map_server/src/map_saver.cpp

```diff
@@ -28,83 +28,191 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "nav2_map_server/map_saver.hpp"
+
 #include <cstdio>
-#include <cstdlib>
-#include <cstring>
-#include <string>
+#include <fstream>
 #include <memory>
+#include <string>
+#include <utility>
+
+#include "Magick++.h"
+#include "nav2_map_server/map_mode.hpp"
+#include "nav_msgs/msg/occupancy_grid.h"
+#include "nav_msgs/srv/get_map.hpp"
 #include "rclcpp/rclcpp.hpp"
-#include "nav2_map_server/map_generator.hpp"
+#include "tf2/LinearMath/Matrix3x3.h"
+#include "tf2/LinearMath/Quaternion.h"
+#include "yaml-cpp/yaml.h"
+
+namespace nav2_map_server
+{
+MapSaver::MapSaver(const rclcpp::NodeOptions & options)
+: Node("map_saver", options), save_next_map_promise{}
+{
+  Magick::InitializeMagick(nullptr);
+  {
+    mapname_ = declare_parameter("output_file_no_ext", "map");
```

Comment:
What do you think about including something unique about the default map name so we're not just constantly overwriting maps in case a user accidentally runs without the parameter. Something like unix time or human readable date-time (or I suppose a random hash)

Reply:
I like that idea, and definitely something to consider for the future. I don’t quite get how people use this tool, and I wanted to keep behavior as close as possible when given the same command line arguments. (otherwise, I’d have changed the default file format and map mode)

Reply:
oh huh, I didn't realize the original `map server` didnt do that, I guess those were tools I built on top of it at some point, my mistake :+1: